src/ol/render/canvas/hitdetect.js

```diff
@@ -68,7 +77,7 @@ export function createHitDetectionImageData(
     for (let j = 0, jj = styles.length; j < jj; ++j) {
       const originalStyle = styles[j];
       const geometry = originalStyle.getGeometryFunction()(feature);
-      if (!geometry || !intersects(extent, geometry.getExtent())) {
+      if (!geometry || !intersects(userExtent, geometry.getExtent())) {
```

Comment:
Would it work to to add `projection` to the argument list (instead of changing `extent` to `userExtent` and adding `userTransform`)?  Then we could use `fromUserExtent(geometry.getExtent(), projection)` here?

Reply:
Thanks for the good suggestion, @tschaub. Definitely looks better now!